flow/src/main/java/flow/Flow.java

```diff
@@ -305,6 +327,14 @@ void enqueue(PendingTraversal pendingTraversal) {
       }
     }
 
+    void bootstrap(History history) {
+      this.nextHistory = checkNotNull(history, "history");
+      if (dispatcher == null) {
+        throw new AssertionError("Bad doExecute method allowed dispatcher to be cleared");
+      }
+      dispatcher.dispatch(new Traversal(null, nextHistory, Direction.REPLACE), this);
```

Comment:
Yay, github is still stupid about comments made directly on sha's, so I'll ask this again:

Why do I want this null? The old code included nextHistory in both slots, which is actually true most of the time, e.g. on config change.